compat/src/internal.d.ts

```diff
@@ -1,9 +1,30 @@
-import { CoreVNode } from '../../src/internal';
-export { PreactElement, ComponentFactory } from '../../src/internal'
-export { ComponentChildren, Ref, ComponentFactory, FunctionalComponent } from '../../src/index';
+{ Ref } from '../../src/index';
+{
+  Component as PreactComponent,
+  VNode as PreactVNode,
+  FunctionalComponent as PreactFunctionalComponent
+} from '../../src/internal';
 
-export interface VNode<T = any> extends CoreVNode<T> {
-	$$typeof: symbol | string;
+export { ComponentChildren } from '../../src/index';
+
+export { PreactElement } from '../../src/internal';
+
+export interface Component<P = {}, S = {}> extends PreactComponent<P, S> {
+  isReactComponent: object;
+  isPureReactComponent?: true;
+}
+
+export interface FunctionalComponent<P = {}> extends PreactFunctionalComponent<P> {
+  shouldComponentUpdate?(nextProps: Readonly<P>): boolean;
+  _forwarded?: true;
 }
 
-export type ForwardFn<T = any, P = {}> = (props: P, ref: Ref<T>) => ComponentFactory<P>;
+export interface VNode<T = any> extends PreactVNode<T> {
+  $$typeof: symbol | string;
```

Comment:
`options.vnode()` is executed in `createVNode()`.
https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/src/create-element.js#L66

https://github.com/developit/preact/blob/fddb5111db985df7551233b8e5ce577b7133fdc9/compat/src/index.js#L160-L165